docs/dev/table/connectors/formats/single-field.md

```diff
@@ -0,0 +1,151 @@
+```
+
+The following creates a table where it reads from (and writes to) the underlying Kafka topic as an anonymous string value by using `single-field` format:
+
+<div class="codetabs" markdown="1">
+<div data-lang="SQL" markdown="1">
+{% highlight sql %}
+CREATE TABLE nginx_log (
+  log STRING
+) WITH (
+  'connector' = 'kafka',
+  'topic' = 'nginx_log',
+  'properties.bootstrap.servers' = 'localhost:9092',
+  'properties.group.id' = 'testGroup',
+  'format' = 'single-field'
+)
+{% endhighlight %}
+</div>
+</div>
+
+Then you can read out the raw data as a pure string, and split it into multiple fields using user-defined-function for further analysing, e.g. `my_split` in the example.
+
+<div class="codetabs" markdown="1">
+<div data-lang="SQL" markdown="1">
+{% highlight sql %}
+SELECT t.hostname, t.datetime, t.url, t.browser, ...
+FROM(
+  SELECT my_split(log) as t FROM nginx_log
+);
+{% endhighlight %}
+</div>
+</div>
+
+In contrast, you can also write a single field of STRING type into Kafka topic as an anonymous string value.
+
+Format Options
+----------------
+
+<table class="table table-bordered">
+    <thead>
+      <tr>
+        <th class="text-left" style="width: 25%">Option</th>
+        <th class="text-center" style="width: 8%">Required</th>
+        <th class="text-center" style="width: 7%">Default</th>
+        <th class="text-center" style="width: 10%">Type</th>
+        <th class="text-center" style="width: 50%">Description</th>
+      </tr>
+    </thead>
+    <tbody>
+    <tr>
+      <td><h5>format</h5></td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Specify what format to use, here should be 'single-field'.</td>
+    </tr>
+    </tbody>
+</table>
+
+Data Type Mapping
+----------------
+
+The table below details the SQL types the format supports, including details of the serializer and deserializer class for encoding and decoding.
+
+<table class="table table-bordered">
+    <thead>
+      <tr>
+        <th class="text-left">Flink SQL type</th>
+        <th class="text-left">Value</th>
+      </tr>
+    </thead>
+    <tbody>
+    <tr>
+      <td><code>CHAR / VARCHAR / STRING</code></td>
+      <td>A UTF-8 encoded text string</td>
+    </tr>
+    <tr>
+      <td><code>BOOLEAN</code></td>
+      <td>A single byte to indicate boolean value, 0 means false, 1 means true.</td>
+    </tr>
+    <tr>
+      <td><code>TINYINT</code></td>
+      <td>A 8-bit signed number</td>
+    </tr>
+    <tr>
+      <td><code>SMALLINT</code></td>
+      <td>A 16-bit signed number</td>
+    </tr>
+    <tr>
+      <td><code>INT</code></td>
+      <td>A 32-bit signed integer</td>
+    </tr>
+    <tr>
+      <td><code>BIGINT</code></td>
+      <td>A 64-bit signed integer</td>
+    </tr>
+    <tr>
+      <td><code>FLOAT</code></td>
+      <td>A 32-bit floating point number</td>
+    </tr>
+    <tr>
```

Comment:
I'm not sure about `RAW` type, because currently it's hard to declare RAW type in DDL.
As an alternative, users can declare it as `BYTES` and use UDF to deserialize the bytes.

Reply:
True, it is difficult to declare the RAW type in DDL. But this is an orthogonal topic that might be solved soon. We should support all data types for our implementations otherwise we need to multiple iterations to get the connectors/formats into a usable state. IMHO adding the RAW here is not a big overhead.